crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -413,52 +431,160 @@ impl<'src> Parser<'src> {
         self.bump(TokenKind::From);
 
         let mut module = None;
-        let mut level = if self.eat(TokenKind::Ellipsis) { 3 } else { 0 };
+        let mut leading_dots = if self.eat(TokenKind::Ellipsis) { 3 } else { 0 };
         let mut progress = ParserProgress::default();
 
         while self.at_ts(DOT_ELLIPSIS_SET) {
             progress.assert_progressing(self);
 
             if self.eat(TokenKind::Dot) {
-                level += 1;
-            }
-
-            if self.eat(TokenKind::Ellipsis) {
-                level += 3;
+                leading_dots += 1;
+            } else if self.eat(TokenKind::Ellipsis) {
+                leading_dots += 3;
             }
         }
 
```

Comment:
Nit: I think I would rewrite this to a loop. It avoids checking twice if the parser is at a DOT or Ellipsis token
```suggestion
        loop {
            progress.assert_progressing(self);

            if self.eat(TokenKind::Dot) {
                leading_dots += 1;
            } else if self.eat(TokenKind::Ellipsis) {
                leading_dots += 3;
            } else {
                break;
            }
        }
```

I otherwise recommend you to use `bump` for the Ellipsis case
```suggestion
        while self.at_ts(DOT_ELLIPSIS_SET) {
            progress.assert_progressing(self);

            if self.eat(TokenKind::Dot) {
                leading_dots += 1;
            } else {
                self.bump(TokenKind::Ellipsis);
                leading_dots += 3;
            }
        }

```